crates/league-toolkit/src/core/meta/property/value/primitives.rs

```diff
@@ -0,0 +1,62 @@
+use crate::core::meta::traits::{PropertyValue, ReadProperty};
+
+macro_rules! impl_prim {
+    ($name:tt, $rust:tt, [$($derive:tt),*], $($method:tt)*) => {
+        #[repr(transparent)]
+        #[derive(Clone, Debug, PartialEq, $($derive),*)]
+        #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+        pub struct $name(pub $rust);
+
+        impl PropertyValue for $name {
+            fn size_no_header(&self) -> usize {
+                core::mem::size_of::<$rust>()
+            }
+        }
+
+        impl ReadProperty for $name {
+            fn from_reader<R: std::io::Read>(
+                reader: &mut R,
+                _legacy: bool,
+            ) -> Result<Self, crate::core::meta::ParseError> {
+                Ok(Self(reader.$($method)*()?))
+            }
+        }
+    };
+}
+
+use crate::core::primitives::Color;
+use crate::util::ReaderExt;
+use byteorder::{ReadBytesExt, LE};
+use glam::{Mat4, Vec2, Vec3, Vec4};
+
+// A "primitive" in this case is just a PropertyValue that just encapsulates
+// a single struct/rust primitive.
+
+impl_prim!(BoolValue, bool, [Eq, Hash], read_bool);
+// TODO (alan): how is this different from a regular BoolValue?
```

Comment:
Afaik this is leftover from before bitfield support was added to league.
This type is also not a primitive, meaning it can't be used as a key for map.